### PR TITLE
Added the Tiberian Sun cyborg logic

### PR DESCRIPTION
**Pull request by Mailaender**
See http://modenc.renegadeprojects.com/Cyborg for reference.

### CODE REVIEW

**Comment by abcdefg30** [comment]
> 21:32 (abcdefg30) ah btw, could we have a damagemodifier on WithPermanentInjury?
21:32 (abcdefg30) actually the damaged cyborg is lying so it should take less damage as with takecover

**Comment by pchote** [comment]
If this granted an upgrade instead of hardcoding a specific modifier(s) then we (and modders) could add any effect that they wanted while damaged.

**Comment by penev92** [comment]
I agree this would be much more useful as `UpgradeOn(Critical)Damage`. We already can change sprites and movement speed on upgrade. We can also change voxels, overlays, weapons, (soon armor, hopefully), and a ton of other stuff.

Currently it is basically hardcoded to do the TS Cyborg behavior. :/

**Comment by abcdefg30** [comment]
Although it was discussed on IRC that we actually want to have an implementation using upgrades,
I'll :+1: / :white_check_mark: this. As @Mailaender argued this can be replaced easily if someone is up to make an upgrade-based implementation and this works for now.

Note to reviewers: This is suffering from #8260.

**Comment by Mailaender** [comment]
I found no evidence at http://modenc.renegadeprojects.com/Cyborg that injured cyborgs take less damage luck prone infantry (shall I really add it?) and I am also not sure if they are really walking slower (shall I remove it?) although that seemed to make sense to me.

**Comment by abcdefg30** [comment]
> I found no evidence at http://modenc.renegadeprojects.com/Cyborg that injured cyborgs take less damage like prone infantry (shall I really add it?)

Yes please, then.

**Comment by Mailaender** [comment]
Removed the speed modifier. Also won't add the damage modifier. I don't intend to re-balance the mod. Also won't implement any upgrade plugs. This should just fix the rendering.

**Comment by LipkeGu** [comment]
SpeedModifider should be go to an extra PR together with the other thing :/

**Comment by chrisforbes** [comment]
:+1:

File: OpenRA.Mods.TS/Traits/WithPermanentInjury.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ * see COPYING.
+ */
+#endregion
+
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.TS.Traits
+{
+	[Desc("Change the sprite and movement speed permanently after a certain amount of damage is taken, even when the hitpoints are regenerated.")]
```
Comment (penev92) [inline OpenRA.Mods.TS/Traits/WithPermanentInjury.cs]: No more "and movement speed".

Comment (Mailaender) [inline OpenRA.Mods.TS/Traits/WithPermanentInjury.cs]: Corrected.

**Comment by SMIFFGIG** [comment]
I think this should be a % chance of them losing their legs. Rather than a hardcoded 100% chance when they reach low health.

Not only does it not make sense that they lose specifically their legs 100% of the time they get damaged. But in the cyborg animation sequence there are multiple death frames for when they cyborg is in standing non prone position (same applies to cyborg commando) which go totally unused in TS.

**Comment by Unit158** [comment]
Adding RNG on a specific unit is a :-1: from me.

**Comment by Unit158** [comment]
I could, however, understand the legs being gibbed by certain weapons.

**Comment by SMIFFGIG** [comment]
Also by making this happen 100% of the time. All of the death animations for the cyborg+cyborg commando will go totally unused.
Seems to make little sense that say a flamethrower could remove the legs of the unit and unfortunate if the specific flame death animation for this unit would also go unused as a consequence.
An oversight from Westwood in the original TS which I hope isn't transferred to OpenRA TS.

**Comment by GraionDilach** [comment]
Could you please open a new issue about this? Think we already discussed this.

**Pull request merged**